File: homework_1/seven_irem.lua
--foo function to print a pair
function foo(k)
     print(k, mytable[k])
end

--initialize
mytable = {["Name"] = "John", ["City"] = "Rome", ["Age"] = 28, ["Height"] =  182}

print(mytable)  -- type and unique id of the object, not the contents

print("\nName is ", mytable["Name"]) --get value for a key
print("Salary is ", mytable["Salary"])   --get value for nonexistent key

print("\n**Initial Table**")
for k, v in pairs(mytable) do --print initial by foo
      foo(k, v)
end

mytable["Weight"] = 79 --add new element

mytable["Age"] = nil   --remove an element
mytable["Salary"] = nil    --remove nonexistent element

mytable["City"] = "Seattle" --modify value

--search for key Name
if mytable["Name"] ~= nil then print("\nkey Name exists") end

--search value Rome
for k, v in pairs(mytable) do
        if v == "Rome" then print("value Rome exists") end
end

--search value Seattle
for k, v in pairs(mytable) do
        if v == "Seattle" then print("value Seattle exists") end
end

print("\n**Table Changed**")
for k, v in pairs(mytable) do --print modified by foo
       foo(k)
end
print("\n")

File: homework_1/seven_irem.rs
use std::collections::HashMap;

fn foo(k: &str, v: &str) {
    println!("Key: {}, Value: {}", k, v); 
}

fn main() { 
    //create
    let mut mymap = HashMap::new();
    //initialize
    mymap.insert("Name", "Kate");
    mymap.insert("City", "Berlin");
    mymap.insert("Age", "27");
    mymap.insert("Height", "159");
    mymap.insert("Weight", "46");
    println!("**Initial Map**");
    for (mapkey, &mapval) in mymap.iter() {
        println!("Key: {}, Value: {}", mapkey, mapval);
    }
    
    println!("\nValue of key Name: {}\n", mymap["Name"]); //get the value for given key
    mymap.insert(&"Country", "Germany"); //adding new element
    mymap.remove(&"City");  //removing element
    *mymap.get_mut("Name").unwrap() = "Alice"; //modify value of Key Name
    
    //checking key existance
    println!("Key City existance: {}", mymap.contains_key(&"City"));
    println!("Key Country existance: {}", mymap.contains_key(&"Country"));
    
    //cheking value existance
    let bool_exist = mymap.values()
        .any(|&val| *val == *"46");
        
    println!("Value 46 Existance: {}", bool_exist);
    
    println!("\n**Modified Map**");
    //printing contents by foo
    for (keyval, &val) in mymap.iter() {
        foo(keyval, val);
    }

}